RPIPs/RPIP-47.md

```diff
@@ -26,34 +26,34 @@ Additionally, the pDAO must have the ability to effect change in ways that benef
-  - Setting the previous `megapool delegate`'s expiration block to occur `delegate_upgrade_buffer` after the upgrade
-- A node operator SHALL be able to upgrade their active `megapool delegate` to the latest `megapool delegate` at any time (including after the active `megapool delegate`'s expiration block)
-- After the active `megapool delegate`'s expiration block:
-  - The active `megapool delegate` SHALL NOT be usable by the node operator
-  - The node operator SHALL NOT be able to claim any rewards
-- The initial setting of `delegate_upgrade_buffer` SHALL be 4 months.
-
-### Megapool recovery delegate
-- There SHALL be a single `megapool recovery delegate` contract that is used by all megapools in the protocol
-- The `megapool recovery delegate` for a megapool SHALL NOT be usable by anyone if the current block is less than or equal to the active `megapool delegate`'s expiration block
-- The `megapool recovery delegate` for a megapool SHALL be usable by any address if the current block is greater than the active `megapool delegate`'s expiration block
-- The `megapool recovery delegate` SHALL have the following capabilities:
-  - Force exiting ALL validators under the megapool
-    - In the Saturn 1 release, this MAY be a non-functional stub
-  - Removal of any pending validators from the Rocket Pool queue
-  - Processing of voluntary exit proofs
-  - Collection of any penalties or deficits
-  - Final distribution of rewards and capital 
-- The `megapool recovery delegate` SHOULD be updated only rarely, and only to support changes to the Ethereum execution layer exit process, or major changes in the Rocket Pool protocol. Simultaneous upgrades to both the `megapool recovery delegate` and the latest `megapool delegate` SHOULD be avoided if possible.
+
+### Upgrade Buffer Enforcement
+- Each `Megapool delegate` contract SHALL have an execution layer `expiration_block` variable which should initialize to "no expiration"
+- Each `Megapool delegate` contract SHALL have a non-modifiable `upgrade_buffer` value.
+  - The `Megapool delegate`'s `upgrade_buffer` SHALL be 120 days.
+- The only permitted modification to a `Megapool delegate`'s `expiration_block` variable SHALL be:
+  - For the pDAO, oDAO, or Security Council to set its value to now + its `upgrade_buffer`
+  - For the pDAO, oDAO, or Security Council to set its value to "no expiration"
+
+### Upgrade
+- When a new `megapool delegate` is released, a contract upgrade SHOULD update all previous `megapool delegate`s `expiration_block`'s to now + their `upgrade_buffer`s
+- A node operator SHALL be able to upgrade their active `megapool delegate` to the latest `megapool delegate` at any time
+- After an active `megapool delegate`'s expiration block has passed:
+  - The expired `megapool delegate` SHALL NOT be usable by the node operator
+  - The node operator SHALL NOT be able to claim any rewards using the expired `megapool delegate`
```

Comment:
Do we want to match the updated wording from Val's recent PR here?

```suggestion
  - Functions in the expired megapool delegate SHALL revert when not called by a protocol contract
  - No capital SHALL be distributed to the withdrawal address(es)```

Reply:
Yes, I'll merge that in, didn't want to double up on it before it was confirmed.